From 34ae59527e710f5fd6acb301677dce9c823896d9 Mon Sep 17 00:00:00 2001
From: Thomas Guillem <thomas@gllm.fr>
Date: Thu, 17 Mar 2022 09:17:09 +0100
Subject: [PATCH 66/77] smb2: always use smb2 timeout

If the smb2_timeout was valid, then not valid (infinite), the last value
was not taken into account.

(cherry picked from commit ac95bf19f9991f506170a3dcb3c57268cb944727)
Signed-off-by: Thomas Guillem <thomas@gllm.fr>
---
 modules/access/smb2.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/modules/access/smb2.c b/modules/access/smb2.c
index 3524e1ab00..12591465fd 100644
--- a/modules/access/smb2.c
+++ b/modules/access/smb2.c
@@ -221,8 +221,6 @@ smb2_service_fd(struct smb2_context *smb2, int fd, int revents)
 static int
 vlc_smb2_mainloop(struct vlc_smb2_op *op)
 {
-    int timeout = -1;
-
     while (op->error_status == 0 && !op->res_done)
     {
         int ret, smb2_timeout;
@@ -236,10 +234,8 @@ vlc_smb2_mainloop(struct vlc_smb2_op *op)
             p_fds[i].events = events;
             p_fds[i].fd = fds[i];
         }
-        if (smb2_timeout != -1)
-            timeout = smb2_timeout;
 
-        if (fds == NULL || (ret = vlc_poll_i11e(p_fds, fd_count, timeout)) < 0)
+        if (fds == NULL || (ret = vlc_poll_i11e(p_fds, fd_count, smb2_timeout)) < 0)
         {
             if (op->log && errno == EINTR)
                 msg_Warn(op->log, "vlc_poll_i11e interrupted");
-- 
2.32.0 (Apple Git-132)

